frontend/src/pages/device/Settings/Editor.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+    <div v-if="deviceOwnerType == 'application'">
+        <form class="space-y-6 max-w-2xl" @submit.prevent>
+            <FormHeading>
+                <template #default>
+                    Editor
+                </template>
+            </FormHeading>
+
+            <!-- Node-RED Version -->
+            <FormRow :error="errors.nodeRedVersion">
+                Node-RED Version
+                <template #description>
+                    Clear this field to use the Node-RED version specified in the device's active snapshot. Defaults to 'latest' if the snapshot does not specify a version.
+                </template>
+                <template #input>
+                    <div class="flex flex-wrap">
+                        <input v-model="input.nodeRedVersion" type="text" class="ff-input ff-text-input w-full" placeholder="latest">
+                    </div>
+                </template>
+            </FormRow>
+            <div class="ff-description !mt-0 max-w-sm italic">NOTE: {{ versionChangeFootnote }}</div>
+
+            <!-- Save -->
+            <ff-button size="small" :disabled="!changed || hasErrors" @click="save">Save Settings</ff-button>
+        </form>
+    </div>
+    <div v-else>
+        Only available to Application bound instances, Instance bound Devices will inherit from the Instance.
+    </div>
+</template>
+
+<script>
+
+import semver from 'semver'
+import { mapState } from 'vuex'
+
+import deviceApi from '../../../api/devices.js'
+import FormHeading from '../../../components/FormHeading.vue'
+import FormRow from '../../../components/FormRow.vue'
+import permissionsMixin from '../../../mixins/Permissions.js'
+import alerts from '../../../services/alerts.js'
+
+export default {
+    name: 'DeviceSettingsEditor',
+    components: {
+        FormHeading,
+        FormRow
+    },
+    mixins: [permissionsMixin],
+    props: {
+        device: {
+            type: Object,
+            required: true
+        }
+    },
+    emits: ['device-updated'],
+    data () {
+        return {
+            input: {
+                nodeRedVersion: ''
+            },
+            errors: {
+                nodeRedVersion: ''
+            },
+            initial: {
+                nodeRedVersion: ''
+            },
+            hasErrors: false
+        }
+    },
+    computed: {
+        ...mapState('account', ['teamMembership']),
+        changed () {
+            const changed = this.initial.nodeRedVersion !== this.input.nodeRedVersion
+            return changed
+        },
+        deviceOwnerType: function () {
+            return this.device?.ownerType || ''
+        },
+        versionChangeFootnote: function () {
+            if (!this.device?.agentVersion || semver.lte(this.device.agentVersion, '2.4.0')) {
+                return 'The device will be updated to the specified version when a different snapshot is deployed'
```

Comment:
```suggestion
                return 'The device will be updated to the specified version the next time a snapshot is deployed'
```

Slight rewording suggestion